src/web/services/statistics-calculator.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@angular/core';
+import { StudentListRowModel } from '../app/pages-instructor/student-list/student-list.component';
+
+/**
+ * The statistics of a course
+ */
+export interface CourseStatistics {
+  numOfSections: number;
+  numOfTeams: number;
+  numOfStudents: number;
+}
+
+/**
+ * Handles the calculation of statistics
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class StatisticsCalculatorService {
+
+  constructor() { }
+
+  /**
+   * Calculates the statistics for a course from a list of students in the course
+   */
+  calculateCourseStatistics(studentList: StudentListRowModel[]): CourseStatistics {
```

Comment:
I think you can put the method into `StudentsService` or `CourseService`.

Reply:
In addition, let's not couple the model data and the business logic (services) together. Can we take `Student[]` as arguments?